### DISCUSSION TITLE
Cant Start 2sxc on Oqtane

### DISCUSSION THREAD

**Discussion by rchibnik** [2sic/2sxc#2482] [Q&A]
Followed the checklist at https://azing.org/2sxc/r/fOG3aByY and watched the video several times.

Copying to the nuget package to my modules folder installs 2sxc but i get an error prior to setup (never get the initial setup page.)

Wondering if I'm skipping a step but have repeated multiple times. I must be missing something. But it should come right up.

Once installed, I created a page called "CMS" and added the content module - the site responds with 

Error: There was an unhandled exception on the current circuit, so this circuit will be terminated. For more details turn on detailed exceptions by setting 'DetailedErrors: true' in 'appSettings.Development.json' or set 'CircuitOptions.DetailedErrors'. 

Don't have an appSettings.Development.json - is this just a copy of appSettings.json with additional settings?

Thanks for any guidance you can provide.

**Comment by rchibnik** [comment]
I should add - I'm using version 2.2.0 of Oqtane

**Comment by iJungleboy** [comment]
You can either copy the appsettings.json and rename, or just make the changes in appsettings.json. 

This will activate more detailed logging.

**Comment by rchibnik** [comment]
12.04 Allows me to install 2sxc on Octane 2.2. Works 
Content templates and Applications wont install, though. Is there a way to install them manually?

I'm running Oqtane in VS2019.

While installing, the VS19 project copies files, but then errors out with:
Error	TS5055	Build:Cannot write file ' ... /Oqtane.Server/wwwroot/Modules/ToSic.Sxc/_/IsXuFQqp/Apps/Content/2sexy/webpack.config.js' because it would overwrite input file.	

I can't re-run oqtane unless i delete the contents of the "_" in the ToSic.Sxc module

Thanks

**Reply by iJungleboy** [reply]
It seems to be having problems with creating temporary files - maybe they are already there.

@tvatavuk could you check if this is an issue? either that older files are not removed and/or something similar?

**Reply by iJungleboy** [reply]
BTW: @rchibnik you can delete the `_` folder manually and try again to keep going, it's temporary stuff only.

**Reply by tvatavuk** [reply]
Hi @rchibnik,
in VS2019 to run Oqtane source code version with 2sxc already installed, please can You in VS2019, in **Oqtane.Server** project "**Exclude From Project**" folders: `wwwroot\Modules\ToSic.Sxc\`, `2sxc`, `Content` and `imageflow_hybrid_cache`?

![image](https://user-images.githubusercontent.com/2724850/132173542-b61c443e-2a59-4083-ae35-2b9d46506688.png)

After this, You should be able to run Oqtane with 2sxc installed in VS2019 without issues.

**Reply by tvatavuk** [reply]
Hi @rchibnik, 
this is a small video that shows [how to install 2sxc 12.04 module in Oqtane 2.2.0 source version](https://www.youtube.com/watch?v=PXVD-gw9EXk).

**Reply by tvatavuk** [reply]
> 12.04 Allows me to install 2sxc on Octane 2.2. Works
> Content templates and Applications wont install, though. Is there a way to install them manually?
> 
> I'm running Oqtane in VS2019.
> 
> While installing, the VS19 project copies files, but then errors out with:
> Error TS5055 Build:Cannot write file ' ... /Oqtane.Server/wwwroot/Modules/ToSic.Sxc/_/IsXuFQqp/Apps/Content/2sexy/webpack.config.js' because it would overwrite input file.
> 
> I can't re-run oqtane unless i delete the contents of the "_" in the ToSic.Sxc module
> 
> Thanks

The main issue is: "**Content templates and Applications wont install, though.**".

RemotePackage method in Installation controller will try to InstallPackage in steps:
1. download 2sxc app from remote server (eg. https://github.com/2sic/app-gallery-fancybox3/releases/download/v06.00.00/2sxcApp_ImageGallery_6.0.0.zip) in temp folder `/Oqtane.Server/wwwroot/Modules/ToSic.Sxc/_/` with _random-temp-name.zip_
2. unzip _random-temp-name.zip_ in temp folder subfolder with random name `/Oqtane.Server/wwwroot/Modules/ToSic.Sxc/_/IsXuFQqp/`
3. import app from `/Oqtane.Server/wwwroot/Modules/ToSic.Sxc/_/IsXuFQqp/` subfolders creating app template folder and files in `/Oqtane.Server/2sxc/...` and `/Oqtane.Server/Content/...` folder structure and import data in SQL server database master tenant
4. delete temporary files/folders `/Oqtane.Server/wwwroot/Modules/ToSic.Sxc/_/IsXuFQqp/`

My understanding from the issue comments is that steps 3 and 4 were not executed, and that leaves 2sxc module without installed apps and with temp files in temp folder. Then vs2019 complains about this temp files on next build, and it can not build until this folders are excluded from the project or deleted.

I can not reproduce this issue in my dev or test env (as also visible in video [how to install 2sxc 12.04 module in Oqtane 2.2.0 source version](https://www.youtube.com/watch?v=PXVD-gw9EXk)).

Hi @rchibnik, please can you check that some external process is not preventing step 3 from happening, like antivirus protection or some windows security permissions that prevents Oqtane code that is running under permissions of the current user, as part of iisexpress running from vs2019 to do its work as expected?

Please can You exclude folders under your `/Oqtane.Server/` from your AV protection and also ensure that your user has full permissions on it (this is usually default behaviour in Oqtane dev env with vs2019 and iisexpress)?

Also, can you enable aspnet core logging, as explained here https://azing.org/oqtane/r/gL2D-i_p and check for more info.
1. For Oqtane source version `web.config` file from this [web.zip](https://github.com/2sic/2sxc/files/7116875/web.zip) save as your `/Oqtane.Server/web.config` file.
2. Create subfolder `/Oqtane.Server/logs/`.
3. Run Oqtane.Server in vs2019 until you get unexpected behaviours. 
4. Analyze log files created (eg. `\Oqtane.Server\logs\stdout_20210906152955_1528.log`) to find more about failures that happens.

**Comment by rchibnik** [comment]
Thank you for the suggestions. The video is particularly helpful. I excluded the recommended folders from my project, but when i install content templates, it causes VS to rebuild the server project, shutting down the oqtane server. There must be another dependency I need to exclude.

**Reply by tvatavuk** [reply]
The good news is that I reproduced this issue in the current **dev branch** of [oqtane.framework](https://github.com/oqtane/oqtane.framework) source code 🤔.

The [latest release of 2sxc module 12.4.0](https://github.com/2sic/2sxc/releases/tag/v12.04.00) is working with the [Oqtane 2.2.0. release version](https://github.com/oqtane/oqtane.framework/releases/tag/v2.2.0) 👌 but not with the current dev branch.

Developer can work with Oqtane source code, but first it need to do `git checkout v2.0.0` before running Oqtane source for the first time in vs2019. Later it can install 2sxc 12.4.0. and after it related Apps and Content.

@rchibnik plase can You check, are you working with oqtane source code with the some not released commit from dev branch or it is v2.2.0 released version of Oqtane source code?

**Reply by rchibnik** [reply]
I always use the release/master branch, rather than the dev; AFAIK I am running 2.2.0 from July 6

I'll retest by getting the latest release branch and reinstalling.

**Reply by rchibnik** [reply]
I dont understand the git checkout - should I run that from the root of my oqtane.framework folder?

**Comment by rchibnik** [comment]
Success!
When I run the oqtane framework from VS (ctrl-f5) it fails - rebuilding midway before the template deployes
When I DEBUG the oqtane framework from VS (F5 or pressing the Play button) it works.

I usually use Ctrl-F5 as its a little faster, and I'm debugging a module in another VS19, rather than debugging code in Oqtane.

Thanks for staying on top of the issue - I look forward to exploring 2SXC.

**Reply by tvatavuk** [reply]
After more testing, I can also confirm that **VS: Ctrl+F5 issue** is related to [VS behaviour](https://forums.asp.net/post/5714435.aspx) when VS file watcher detects that new files are in web app for compilation (eg. 2sxc Apps or Content templates), iis express quits and vs restarts it again. 

Util like https://github.com/jexuswebserver/jexusmanager can be used to start Oqtane.Server IISEXPRESS web site without VS: Ctr + F5. In this case, 2sxc Apps and Content download and installation workflow work as expected. The same is when we use full IIS to directly host Oqtane.Server website, also 2sxc Apps and Content download and installation workflow work as expected.

Currently, I do not see anything in 2sxc code that we can change to fix this VS: Ctrl + F5 issue. Even moving 2sxc temp folder "_" from wwwroot to path out of web site is not helping because VS: Ctrl + F5 will still restart iisexpress and that breaks 2sxc install app or content template workflow. 

The solution is to use VS: F5, or other Oqtane.Server website running options mentioned.
